Embrace Unity iOS Interface/ExposedAPI.swift

```diff
@@ -0,0 +1,527 @@
+    guard let event else {
+        return
+    }
+    
+    if let _event = String(validatingUTF8: event) {
+        EmbraceManager.addBreadCrumb(event: _event)
+    }
+}
+
+@_cdecl("set_username")
+public func set_username(username: UnsafePointer<CChar>?) {
+    guard let username else {
+        return
+    }
+    
+    if let _username = String(validatingUTF8: username) {
+        EmbraceManager.setUsername(userName: _username)
+    }
+}
+
+@_cdecl("clear_username")
+public func clear_username() {
+    EmbraceManager.clearUsername()
+}
+
+@_cdecl("set_user_email")
+public func set_user_email(email: UnsafePointer<CChar>?) {
+    guard let email else {
+        return
+    }
+    
+    if let _email = String(validatingUTF8: email) {
+        EmbraceManager.setUserEmail(userEmail: _email)
+    }
+}
+
+@_cdecl("clear_user_email")
+public func clear_user_email(email: UnsafePointer<CChar>?) {
+    EmbraceManager.clearUserEmail()
+}
+
+@_cdecl("add_user_persona")
+public func add_user_persona(persona: UnsafePointer<CChar>?) {
+    guard let persona else {
+        return
+    }
+    
+    if let _persona = String(validatingUTF8: persona) {
+        EmbraceManager.addUserPersona(persona: _persona)
+    }
+}
+
+@_cdecl("clear_user_persona")
+public func clear_user_persona(persona: UnsafePointer<CChar>?) {
+    guard let persona else {
+        return
+    }
+    
+    if let _persona = String(validatingUTF8: persona) {
+        EmbraceManager.clearUserPersona(persona: _persona)
+    }
+}
+
+@_cdecl("clear_all_user_personas")
+public func clear_all_user_personas() {
+    EmbraceManager.clearAllUserPersonas()
+}
+
+@_cdecl("add_session_property")
+public func add_session_property(key: UnsafePointer<CChar>?, value: UnsafePointer<CChar>?, permanent: Bool) -> Bool {
+    guard let key, let value else {
+        return false
+    }
+    
+    if let _key = String(validatingUTF8: key), let _value = String(validatingUTF8: value) {
+        return EmbraceManager.addSessionProperty(key: _key, value: _value, permanent: permanent)
+    }
+    
+    return false
+}
+
+@_cdecl("remove_session_property")
+public func remove_session_property(key: UnsafePointer<CChar>?) {
+    guard let key else {
+        return
+    }
+    
+    if let _key = String(validatingUTF8: key) {
+        EmbraceManager.removeSessionProperty(key: _key)
+    }
+}
+
+@_cdecl("log_message_with_severity_and_properties")
+public func log_message_with_severity_and_properties(message: UnsafePointer<CChar>?, severity: UnsafePointer<CChar>?, propsJson: UnsafePointer<CChar>?) {
+    
+    guard let message, let severity else {
+        return
+    }
+    
+    if let _message = String(validatingUTF8: message), 
+        let _severity = String(validatingUTF8: severity) {
+        EmbraceManager.logMessageWithSeverityAndProperties(message: _message,
+                                                           severity: _severity,
+                                                           properties: 
```

Comment:
for error and warning logs should the current stack trace be passed through? We pass this through on RN but not sure if there's a similar requirement on Unity: https://github.com/embrace-io/embrace-react-native-sdk/blob/main/packages/core/src/index.ts#L151

Reply:
We have not previously on Unity for iOS 5.x, and I anchored to previous behavior in these cases.

However, there's a really good argument for doing so and I think it's preferable to send a stack trace in retrospect. The only issue there is that the current public facing API for this doesn't expose a place for the user to easily submit a stack trace as part of their log and has not traditionally. That would imply a breaking change, which I tended to avoid on this migration.

Reply:
sounds good, ya if it wasn't part of previously functionality I think reasonable to also not include here and leave for a future version